include/boost/hana/fwd/map.hpp

```diff
@@ -280,6 +280,71 @@ BOOST_HANA_NAMESPACE_BEGIN
         return tag-dispatched;
     };
 #endif
+
+    //! Returns the intersection of two maps.
+    //! @relates hana::map
+    //!
+    //! Given two maps `xs` and `ys`, `intersection(xs, ys)` is a new map
+    //! containing exactly those (key, value) pairs from xs, for which key
+    //! is present in `ys`.
+    //! In other words, the following holds for any object `pair(k, v)`:
+    //! @code
+    //!     pair(k, v) ^in^ intersection(xs, ys) if and only if (k, v) ^in^ xs && k ^in^ ys.keys()
+    //! @endcode
+    //!
+    //!
+    //! @note
+    //! This function is not commutative, i.e. `intersection(xs, ys)` is
+    //! not necessarily same as `intersection(ys, xs)`. Although, set of
+    //! keys of `intersection(xs, ys)` and set of keys of `intersection(ys, xs)`
+    //! are always same. `intersection(xs, ys)` contains values present in xs,
+    //! for a given key, and `intersection(ys, xs)` contains values present
+    //! in ys for a given key.
+    //!
+    //!
+    //! @param xs, ys
+    //! Two maps to intersect.
+    //!
+    //!
+    //! Example
+    //! -------
+    //! @include example/map/intersection.cpp
+#ifdef BOOST_HANA_DOXYGEN_INVOKED
+    constexpr auto intersection = [](auto&& xs, auto&& ys) {
+        return tag-dispatched;
+    };
+#endif
+
+    //! Returns the difference of two maps.
+    //! @relates hana::map
+    //!
+    //! Given two maps `xs` and `ys`, `difference(xs, ys)` is a new map
+    //! containing exactly those (key, value) pairs from xs, for which key
+    //! is not present in `keys(ys)`.
+    //! In other words, the following holds for any object `pair(k, v)`:
+    //! @code
+    //!     pair(k, v) ^in^ difference(xs, ys) if and only if (k, v) ^in^ xs && k ^not in^ ys.keys()
+    //! @endcode
+    //!
+    //!
+    //! @note
+    //! This function is not commutative, i.e. `difference(xs, ys)` is
+    //! not necessarily same as `difference(ys, xs)`.
+    //!
+    //!
+    //! @param xs, ys
+    //! Two maps to find difference.
```

Comment:
Please change to: "Two maps to compute the difference of.". I'm using this formulation for `union_`; it might not be perfect but it'll be consistent.